1-select-tasks/4.txt

```diff
@@ -2,4 +2,11 @@ Create a query to return:
 
 | Customer Id | Total number of Orders | % of all orders |
  
-order data by % - higher percent at the top.
\ No newline at end of file
+order data by % - higher percent at the top.
+
+--task4
+SELECT [o].[CustomerID], COUNT([o].[CustomerID]) AS [Total number of Orders],
+ROUND(CAST(COUNT([o].[OrderID]) AS NUMERIC) *100/SUM(COUNT([o].[OrderID])) OVER(), 3) AS [ % of all orders]
```

Comment:
It is possible to drop the Cast, see decimal constants (https://docs.microsoft.com/en-us/sql/t-sql/data-types/constants-transact-sql#decimal-constants).  You could drop round func as well since it wasn't part of the task. Like the solution with over clause